build/yamls/base/crds.yml

```diff
@@ -434,9 +434,51 @@ spec:
                     # Ensure that Spec.AppliedTo does not allow IPBlock field
                     properties:
                       podSelector:
-                        x-kubernetes-preserve-unknown-fields: true
+                        type: object
+                        properties:
+                          matchExpressions:
+                            type: array
+                            items:
+                              type: object
+                              properties:
+                                key:
+                                  type: string
+                                operator:
+                                  enum:
+                                    - In
+                                    - NotIn
+                                    - Exists
+                                    - DoesNotExist
+                                  type: string
+                                values:
+                                  type: array
+                                  items:
+                                    type: string
+                          matchLabels:
+                            type: object
```

Comment:
I guess you need to set "x-kubernetes-preserve-unknown-fields: true" for matchLabels, otherwise all keys are pruned so the e2e tests failed. Glad to see it's caught by the tests.
https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#controlling-pruning

Reply:
Thanks for advice，obviously，some resource can not created as expected in e2e test cases cause the fails, but I have verified those crds in my local kind cluster; I reviewed the e2e test cases maybe found sth, waiting for verifing, however I did not find an efficient method to implement the e2e test in my local dev env. Hope you are right and will solve the problem as soon as possible.